compiler/model-import-verifier/src/EvalDriver.cpp

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <luci/Importer.h>
+#include <luci_interpreter/Interpreter.h>
+#include <luci/CircleExporter.h>
+#include <luci/CircleFileExpContract.h>
+
+#include <cstdlib>
+#include <fstream>
+#include <vector>
+#include <string>
+#include <numeric>
+#include <random>
+
+#include "sample_model.h"
+
+namespace
+{
+
+std::unique_ptr<luci::Module> import_model_file(const std::string &filename, bool copy_const)
+{
+  std::ifstream fs(filename, std::ifstream::binary);
+  if (fs.fail())
+  {
+    throw std::runtime_error("Cannot open model file \"" + filename + "\".\n");
+  }
+
+  // define model's size
+  uint32_t model_size = 0;
+  {
+    fs.seekg(0, fs.end);
+    model_size = fs.tellg();
+    fs.seekg(0, fs.beg);
+  }
+
+  // allocate memory and provide memory management to shared ptr
+  std::shared_ptr<uint8_t[]> model_data(new uint8_t[model_size]);
+  std::copy(std::istreambuf_iterator<char>(fs), std::istreambuf_iterator<char>(), model_data.get());
+
+  auto const pointer_to_pass = copy_const ? nullptr : model_data;
+  return luci::Importer(pointer_to_pass).importModule(circle::GetModel(model_data.get()));
+}
+
+std::unique_ptr<luci::Module> import_model_constant_buffer(const uint8_t *buffer, bool copy_const)
+{
+  // wrap constant buffer to shared ptr with deprecated deletion
+  auto const nothing_to_do_deleter = [](uint8_t *) { /* Do nothing */ };
+  std::shared_ptr<uint8_t[]> model_data(const_cast<uint8_t *>(buffer), nothing_to_do_deleter);
```

Comment:
Wrap to `shared_ptr` and use empty deleter for models provided as const pointer to buffer.